src/panel/direction/DirectionPanel.jsx

```diff
@@ -92,6 +95,16 @@ export default class DirectionPanel extends React.Component {
     Promise.all(poiRestorePromises).then(([ origin, destination ]) => {
       persistentPointState.origin = origin;
       persistentPointState.destination = destination;
+
+      // Show the raw POI name/latlon while waiting for computeRoutes to complete
+      if (origin) {
+        this.setState({ originInputText: origin.name });
```

Comment:
Even if React can combine some operations behind the scene, it's better to regroup `setState` calls where it's possible, to limit useless re-renders.
Here, this can be grouped with the existing `setState` below.